test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerUtilityTests.cs

```diff
@@ -0,0 +1,153 @@
+                        ""net472"": {
+                            ""dependencies"": {
+                                ""b"" : ""2.0.0"",
+                                ""a"" : ""1.0.0"" 
+                            }
+                        },
+                        ""net5.0"": {
+                            ""dependencies"": {
+                                ""b"" : ""2.0.0"",
+                            }
+                        }
+                    }
+                }";
+            Dictionary<NuGetFramework, string> originalFrameworks = new()
+            {
+                { FrameworkConstants.CommonFrameworks.Net472, "net472" },
+                { FrameworkConstants.CommonFrameworks.Net50, "net50" }
+            };
+
+            var originalPackageSpec = ProjectTestHelpers.GetPackageSpecWithProjectNameAndSpec("project", @"C:\", referenceSpec);
+
+            var buildIntegrationInstallationContext = NuGetPackageManager.CreateInstallationContextForPackageId(packageIdentityId: "a", originalPackageSpec, unsuccessfulFrameworks: new(), originalFrameworks);
+            buildIntegrationInstallationContext.OriginalFrameworks.Should().Equal(originalFrameworks);
+            buildIntegrationInstallationContext.SuccessfulFrameworks.Should().HaveCount(1);
+            buildIntegrationInstallationContext.UnsuccessfulFrameworks.Should().HaveCount(1);
+            buildIntegrationInstallationContext.UnsuccessfulFrameworks.Should().Contain(FrameworkConstants.CommonFrameworks.Net50);
+            buildIntegrationInstallationContext.SuccessfulFrameworks.Should().Contain(FrameworkConstants.CommonFrameworks.Net472);
+        }
+
+        [Fact]
+        public void CreateInstallationContextForPackageId_WithFailedConditionalOperation_ReturnsCorrectValue()
+        {
+            string referenceSpec = @"
+                {
+                    ""frameworks"": {
+                        ""net472"": {
+                            ""dependencies"": {
+                                ""b"" : ""2.0.0"",
+                                ""a"" : ""1.0.0"" 
+                            }
+                        },
+                        ""net5.0"": {
+                            ""dependencies"": {
+                            }
+                        }
+                    }
+                }";
+            Dictionary<NuGetFramework, string> originalFrameworks = new()
+            {
+                { FrameworkConstants.CommonFrameworks.Net472, "net472" },
+                { FrameworkConstants.CommonFrameworks.Net50, "net50" }
+            };
+
+            var originalPackageSpec = ProjectTestHelpers.GetPackageSpecWithProjectNameAndSpec("project", @"C:\", referenceSpec);
+
+            var buildIntegrationInstallationContext = NuGetPackageManager.CreateInstallationContextForPackageId(packageIdentityId: "a", originalPackageSpec, unsuccessfulFrameworks: new() { FrameworkConstants.CommonFrameworks.Net472 }, originalFrameworks);
```

Comment:
It's part of the test name. 
A package is installed conditionally, but the update to that conditional package failed, hence net472.

Reply:
Thanks for the clarification; I now understand. Essentially, the test is to verify that if an update fails for the `net472` framework, then the collection of unsuccessful frameworks should include both `net5`, since it lacks dependencies on package 'a', and 'net472', because the update failed for some reason. Is my understanding correct?

Reply:
yes